log/context.go

```diff
@@ -48,7 +48,7 @@ func WithRequestID(ctx context.Context, requestID string, fields ...LoggableFiel
 // It can be used when there isn't a single, clear, unique id associated with a request (e.g.,
 // a block or tx hash).
 func WithNewRequestID(ctx context.Context, fields ...LoggableField) context.Context {
-	return WithRequestID(ctx, uuid.New().String(), fields...)
+	return WithRequestID(ctx, limit(uuid.New().String(), 10), fields...)
```

Comment:
NIT: this does the same thing without an extra utility function:

```suggestion
	return WithRequestID(ctx, fmt.Sprintf("%.10s", uuid.New()), fields...)
```